File: Lista de Exercicio Paradigmas/Lista Python/Ex5.py
class Animal:
    def som(self):
        raise NotImplementedError("Este método deve ser implementado por subclasses.")

class Cachorro(Animal):
    def som(self):
        return "Au Au!"

class Gato(Animal):
    def som(self):
        return "Miau!"

def fazer_som_animais(animais):
    for animal in animais:
        print(f'O animal faz: {animal.som()}')

animais = [Cachorro(), Gato(), Cachorro(), Gato()]

fazer_som_animais(animais)

File: Lista de Exercicio Paradigmas/Lista Java/Ex5.java
abstract class Animal {
    abstract String som();
}

class Cachorro extends Animal {
    @Override
    String som() {
        return "Au Au!";
    }
}

class Gato extends Animal {
    @Override
    String som() {
        return "Miau!";
    }
}

public class Main {
    public static void fazerSomAnimais(Animal[] animais) {
        for (Animal animal : animais) {
            System.out.println("O animal faz: " + animal.som());
        }
    }

    public static void main(String[] args) {
        Animal[] animais = {new Cachorro(), new Gato(), new Cachorro(), new Gato()};

        fazerSomAnimais(animais);
    }
}
